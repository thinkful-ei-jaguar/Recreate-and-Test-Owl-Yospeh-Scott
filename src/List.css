.List {
  position: absolute;
  top: 0;
  left: 0;
  width: 25vw;
  height: 85vh;
  background-color: thistle;
  overflow: auto;
}