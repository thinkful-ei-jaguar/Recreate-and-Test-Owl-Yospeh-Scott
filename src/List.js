import React from 'react';
import Participant from './Participant'
import './List.css';

export default function List(props) {
  return (
    <section className='List'>
      <div className='List-participants'>
        {props.cards.map((participant) =>  
          <Participant
            key={participant.id}
            name={participant.name}
            avatar={participant.avatar}
            inSession={participant.inSession}
            onStage={participant.onStage}
          />
        )}
      </div>
    </section>
  )
}
