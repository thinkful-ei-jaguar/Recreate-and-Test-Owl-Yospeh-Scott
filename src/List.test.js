import React from 'react';
import ReactDOM from 'react-dom';
import List from './List'
import renderer from 'react-test-renderer';


describe('List Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <List 
        cards={[
          {id: 1, name: 'Yosef', avatar: 'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1', inSession: true, onStage: true },
          {id: 2, name: 'Scott', avatar: 'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1', inSession: false, onStage: true }
         ]}
      />, 
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<List 
        header='List 1'
        cards={[
          {id: 1, title: 'Raindrops', content: 'jdnsda'},
          {id: 2, title: 'Whiskers', content: 'hjkshfs'}
         ]}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});