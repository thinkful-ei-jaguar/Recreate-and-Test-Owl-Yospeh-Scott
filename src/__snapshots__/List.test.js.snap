// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders the UI as expected 1`] = `
<section
  className="List"
>
  <div
    className="List-participants"
  >
    <div
      className="participant"
    >
      <img
        alt="avatar"
        className="avatar"
      />
      <div
        className="information"
      >
        <p />
        <p />
        <p />
      </div>
    </div>
    <div
      className="participant"
    >
      <img
        alt="avatar"
        className="avatar"
      />
      <div
        className="information"
      >
        <p />
        <p />
        <p />
      </div>
    </div>
  </div>
</section>
`;
